src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-form/rbd-form.component.ts

```diff
@@ -0,0 +1,454 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+
+import * as _ from 'lodash';
+
+import { PoolService } from '../../../shared/api/pool.service';
+import { NotificationType } from '../../../shared/enum/notification-type.enum';
+import { FinishedTask } from '../../../shared/models/finished-task';
+import { DimlessBinaryPipe } from '../../../shared/pipes/dimless-binary.pipe';
+import { FormatterService } from '../../../shared/services/formatter.service';
+import { NotificationService } from '../../../shared/services/notification.service';
+import { RbdService } from '../../../shared/services/rbd.service';
+import { TaskManagerMessageService } from '../../../shared/services/task-manager-message.service';
+import { TaskManagerService } from '../../../shared/services/task-manager.service';
+import { RbdFormCreateRequestModel } from './rbd-form-create-request.model';
+import { RbdFormEditRequestModel } from './rbd-form-edit-request.model';
+import { RbdFormResponseModel } from './rbd-form-response.model';
+
+@Component({
+  selector: 'cd-rbd-form',
+  templateUrl: './rbd-form.component.html',
+  styleUrls: ['./rbd-form.component.scss']
+})
+export class RbdFormComponent implements OnInit {
+
+  rbdForm: FormGroup;
+  featuresFormGroups: FormGroup;
+  defaultFeaturesFormControl: FormControl;
+  deepFlattenFormControl: FormControl;
+  layeringFormControl: FormControl;
+  exclusiveLockFormControl: FormControl;
+  objectMapFormControl: FormControl;
+  journalingFormControl: FormControl;
+  fastDiffFormControl: FormControl;
+
+  pools: Array<string> = null;
+  dataPools: Array<string> = null;
+  features: any;
+  featuresList = [];
+
+  routeParamsSubscribe: any;
+  pool: string;
+
+  advancedEnabled = false;
+
+  editing = false;
+
+  response: RbdFormResponseModel;
+
+  defaultObjectSize = '4MiB';
+
+  constructor(private route: ActivatedRoute,
+              private router: Router,
+              private poolService: PoolService,
+              private rbdService: RbdService,
+              private formatter: FormatterService,
+              private dimlessBinaryPipe: DimlessBinaryPipe,
+              private taskManagerService: TaskManagerService,
+              private taskManagerMessageService: TaskManagerMessageService,
+              private notificationService: NotificationService) {
+    this.features = {
+      'deep-flatten': {
```

Comment:
Nit: it looks like I can check an unchecked ```Deep flatten``` button on the edit pane if it wasn't already enabled.

Reply:
... with the latest update I cannot disable deep-flatten on images where it's already enabled (it should be disable-only).